libraries/dagster-contrib-weaviate/dagster_contrib_weaviate_tests/test_resource.py

```diff
@@ -0,0 +1,188 @@
+    )
+
+    question_objs = list()
+    for i, d in enumerate(data_vector):
+        question_objs.append(
+            weaviate.classes.data.DataObject(
+                properties={
+                    "answer": d["Answer"],
+                    "question": d["Question"],
+                    "category": d["Category"],
+                },
+                vector=d["vector"],
+            )
+        )
+
+    questions = client.collections.get("Question")
+    questions.data.insert_many(question_objs)
+
+    time.sleep(
+        1
+    )  # Weaviate recommends a short sleep here, due to async operations running in the background.
+
+    return client
+
+
+def test_local_resource():
+    """Starts up an embedded instance of Weaviate, adds some data, and then queries it.
+    To avoid installing heavy models on github-actions (which could incur costs),
+    we use the "bring-your-own-vectors" method,
+    as described here: https://weaviate.io/developers/weaviate/starter-guides/custom-vectors
+    """
+
+    @asset
+    def query_local_weaviate_asset(weaviate_local: WeaviateResource):
+        with create_embedded_weaviate_and_add_data():
+            with weaviate_local.get_client() as client:
+                questions = client.collections.get("Question")
+                response = questions.query.near_vector(
+                    near_vector=read_query_vector_from_file(),
+                    limit=2,
+                    return_metadata=weaviate.classes.query.MetadataQuery(
+                        certainty=True
+                    ),
+                )
+
+                assert len(response.objects) == 2
+
+    result = materialize(
+        [query_local_weaviate_asset],
+        resources={
+            "weaviate_local": WeaviateResource(
+                connection_config=LocalConfig(
+                    host="localhost", port=WEAVIATE_PORT, grpc_port=WEAVIATE_GRPC_PORT
+                )
+            )
+        },
+    )
+
+    assert result.success
+
+
+WCD_URL = "https://rand0md1g1t5.something.weaviate.cloud"
+WCD_APIKEY = "FakeWeaviateCloudApiKey"
+COHERE_APIKEY = "FakeCohereApiKey"
+
+# Required to avoid an infinite recursion
+# (since our mock needs to forward some requests to the original httpx.Send)
+real_httpx_send_method = httpx.AsyncClient.send
+
+
+# this method only returns a mock response if the URL accessed is weaviate.
+# Otherwise, it passes the request through to real_httpx_send_method
+# Impotant since Weaviate might make some PyPI requests (which don't need to be mocked).
+async def mock_httpx_send_method(self, *args, **kwargs):
+    request = args[0]
+
+    if "weaviate.cloud" not in str(request.url):
+        return await real_httpx_send_method(self, *args, **kwargs)
+
+    assert WCD_URL in str(request.url)
+    assert ("x-cohere-api-key", COHERE_APIKEY) in request.headers.items()
+    assert WCD_APIKEY in request.headers["authorization"]
+
+    result = httpx.Response(
```

Comment:
please indicate where the mocked response came from so that we can replicate/change this in the future